docs/pages/guides/schema-validation.mdx

```diff
@@ -0,0 +1,120 @@
+---
+
+## Why schema validation?
+Schema validation is essential for future-proofing your application and ensuring that implementing new features or data structures into your app will keep the integration you have established with Liveblocks. Our storage block is flexible in that you can store different Type structures depending on your unique needs. What happens if you alter some (or all) of a storage object’s properties but have existing data?
+
+The issue is that storage with Liveblocks contains a different type structure than your application (client)- which could break your application. How do you avoid this situation? By enforcing schema validation. While optional, schema validation is a vital tool for applications in a production environment- it ensures that errors like the one described above can be avoided and resolved while working in development mode. We’ll elaborate a bit more on the inner workings of schema validation below.
+
+## How schema validation works
+As described above, the primary purpose of schema validation is to prevent corrupted data from being loaded from the server. Two things need to happen to keep our data in sync and eliminate the risk of errors. First, we must ensure that data loaded into a room matches the underlying storage Tree and defined schema (if it exists). Second, we will introduce the rule that each mutation that occurs to their storage data will either be accepted or rejected.  
+
+When a room loads, it determines whether a schema exists; if it does, the value of the schema is parsed and assigned to the room.
+
+<!-- Insert an animation here !-->
+
+If the room does not enforce schema validation, all mutations in the room are accepted. 
+For example, imagine that a client performs an operation that updates the storage tree: regardless of whether or not this could break the application, Liveblocks has no opinion on what this mutation represents and accepts the incoming operation and synchronizes it with other clients. There is no right or wrong way to mutate data. After the operation is completed, the updated storage tree is forwarded to other users in the room. 
+
+When a room enforces schema validation and the client updates values in the storage, the operation is sent to the Liveblocks server, verifying that the updated Storage Tree is consistent with the defined schema attached to the room. If the verification succeeds, the storage data gets updated and synchronized with other users (clients) in the room. If the schema validation fails, the mutation gets rejected: storage will not be updated, other clients will never see the mutation, and the client that made the mutation will throw an error, and the developer can fix it. Situations like this should only happen in development mode, and the issue should be fixed by the developer, similar to fixing a TypeScript error before publishing it to production.
+
+## Create a schema
+To create a schema, you will need to POST to the schemas endpoint:
```

Comment:
Now that we have a basic dashboard, I would update this section to show the flow through the UI and also indicate that it's possible to use the API with a link to our[ HTTP endpoints ](https://liveblocks.io/docs/api-reference/rest-api-endpoints).

Reply:
Will do :)

Reply:
I think we discussed but are we adding the documentation to http endpoints before launch? I saw a story assigned to me surrounding this, but was shown as something that would occur later